receiver/prometheusreceiver/metrics_receiver_target_allocator_test.go

```diff
@@ -497,6 +497,10 @@ func TestTargetAllocatorJobRetrieval(t *testing.T) {
 
 			require.NoError(t, receiver.Start(ctx, componenttest.NewNopHost()))
 
+			t.Cleanup(func() {
```

Comment:
Is this related to the change, or just a useful fix?

Reply:
This is from running all the tests with the UI enabled by default. Some tests are not cleaning up before the next one runs so the port is already taken on the next test. I could also remove these changes since it's not enabled by default and has separate tests for the UI